src/cacao.ts

```diff
@@ -1,16 +1,19 @@
 import { verifyMessage } from '@ethersproject/wallet'
 import * as dagCbor from '@ipld/dag-cbor'
+import { verify } from '@stablelib/ed25519'
+import { AccountId } from 'caip'
 import * as multiformats from 'multiformats'
 import * as Block from 'multiformats/block'
 import { sha256 as hasher } from 'multiformats/hashes/sha2'
-import { SiweMessage } from './siwe.js'
-import { AccountId } from 'caip'
+import { fromString as u8aFromString } from 'uint8arrays/from-string'
+import { SiweMessage } from './siwx/siwe.js'
+import { SiwsMessage } from './siwx/siws.js'
 
 // 5 minute default clockskew
 const CLOCK_SKEW_DEFAULT_SEC = 5 * 60
 
 export type Header = {
-  t: 'eip4361'
+  t: 'eip4361' | 'CASAXX'
```

Comment:
I don't think using `CASAXX` here makes sense. The type here should fully describe how to map the data to the payload that is being signed over. This will be different for every blockchain I assume? Maybe we call the one for solana `CAIPXXSol`?

Reply:
sure!

Reply:
does this imply we also change `eip4361` to be `eip155-caip122`? though that might be a breaking change for those using ethereum CACAOs on mainnet. not sure if Gitcoin Passports is doing that?

Reply:
I don't think so `eip4361` sufficiently describes what's going on. I don't think we need to follow a specific naming convention here. We just need to make sure we represent the alg used to convert the encoded CACAO to the bytestring that is signed over.

Reply:
Actually, discussed this with @ukstv - we can just keep this as `caip122` because the payload does not change regardless of what blockchain is used, given the SIWx data model remains the same.

Reply:
That should not matter, as CAIP-122 === SIWx. How signing/verification works is dependent on the namespace spec, but the data payload remains the same which conforms to CAIP-122. i.e. if the payload structure is the same, we're signing over caip122. the signature type etc can change dependent on namespace spec.